bigtable/google/cloud/bigtable/cluster.py

```diff
@@ -112,7 +172,25 @@ def __ne__(self, other):
 
     def reload(self):
         """Reload the metadata for this cluster."""
-        self._instance._client.instance_admin_client.get_cluster(self.name)
+        cluster_pb = self._instance._client.instance_admin_client.get_cluster(
+            self.name)
+
+        # NOTE: _update_from_pb does not check that the project and
+        #       cluster ID on the response match the request.
+        self._update_from_pb(cluster_pb)
+
+    def exists(self):
+        """Check whether the cluster already exists.
+
+        :rtype: bool
+        :returns: True if the table exists, else False.
+        """
+        client = self._instance._client
+        try:
+            client.instance_admin_client.get_cluster(name=self.name)
+            return True
+        except exceptions.NotFound:
+            return False
```

Comment:
Can you please add a comment that there could be other exceptions that are returned to the user? I guess we never explicitly document grpc/gax exceptions, but maybe we should.